tapgarden/seedling.go

```diff
@@ -183,10 +183,49 @@ func (c Seedling) validateFields() error {
 func (c Seedling) validateGroupKey(group asset.AssetGroup,
 	anchorMeta *proof.MetaReveal) error {
 
-	// We must be able to sign with the group key.
-	if !group.GroupKey.IsLocal() {
-		groupKeyBytes := c.GroupInfo.GroupPubKey.SerializeCompressed()
-		return fmt.Errorf("can't sign with group key %x", groupKeyBytes)
+	switch {
```

Comment:
Instead of this switch you can just return early with:
```golang
if c.ExternalKey.IsNone() && !group.GroupKey.IsLocal() {
	groupPubKey := c.GroupInfo.GroupPubKey
	groupKeyBytes := groupPubKey.SerializeCompressed()
	return fmt.Errorf("can't sign with group key %x",
		groupKeyBytes)
}
```

And you don't need to wrap the `fn.MapOptionZ` call in `case c.ExternalKey.IsSome():`. Because `fn.MapOptionZ` will just return `err == nil` if `c.ExternalKey.IsNone()`. Maybe we can save on indentation like that.

Reply:
Ah nice, that's actually much nicer. Updated, thanks.